From 512e77f66fa1f82a54f3637a58234ba2d622183e Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Tue, 9 Aug 2016 20:37:36 +0200
Subject: [PATCH] Fixed endless recursion crashes caused by summary screen
 handling cleanup TODO: Gradualy remove in some old screens the duplicated
 onChangedEntry handlers

---
 usr/lib/enigma2/python/Components/ConfigList.py | 6 +++---
 usr/lib/enigma2/python/Screens/Setup.py         | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/usr/lib/enigma2/python/Components/ConfigList.py b/usr/lib/enigma2/python/Components/ConfigList.py
index 43b553d..35fc3af 100644
--- a/usr/lib/enigma2/python/Components/ConfigList.py
+++ b/usr/lib/enigma2/python/Components/ConfigList.py
@@ -183,15 +183,15 @@ class ConfigListScreen(object):
 		self["config"] = ConfigList(list, session = session)
 		self.setup_title = ""
 
-		self.onChangedEntry = []
+		self.onConfigEntryChanged = []
 		if on_change:
-			self.onChangedEntry.append(on_change)
+			self.onConfigEntryChanged.append(on_change)
 
 		if not self.handleInputHelpers in self["config"].onSelectionChanged:
 			self["config"].onSelectionChanged.append(self.handleInputHelpers)
 
 	def _changedEntry(self):
-		for fnc in self.onChangedEntry:
+		for fnc in self.onConfigEntryChanged:
 			fnc()
 
 	def handleInputHelpers(self):
diff --git a/usr/lib/enigma2/python/Screens/Setup.py b/usr/lib/enigma2/python/Screens/Setup.py
index d8d4f99..52e9a74 100644
--- a/usr/lib/enigma2/python/Screens/Setup.py
+++ b/usr/lib/enigma2/python/Screens/Setup.py
@@ -30,19 +30,19 @@ class SetupSummary(Screen):
 
 	def __init__(self, session, parent):
 		Screen.__init__(self, session, parent = parent)
-		self["SetupTitle"] = StaticText(_(parent.setup_title) or parent["Title"].text)
+		self["SetupTitle"] = StaticText(_(parent.setup_title) if parent.setup_title else parent["Title"].text)
 		self["SetupEntry"] = StaticText("")
 		self["SetupValue"] = StaticText("")
 		self.onShow.append(self.addWatcher)
 		self.onHide.append(self.removeWatcher)
 
 	def addWatcher(self):
-		self.parent.onChangedEntry.append(self.selectionChanged)
+		self.parent.onConfigEntryChanged.append(self.selectionChanged)
 		self.parent["config"].onSelectionChanged.append(self.selectionChanged)
 		self.selectionChanged()
 
 	def removeWatcher(self):
-		self.parent.onChangedEntry.remove(self.selectionChanged)
+		self.parent.onConfigEntryChanged.remove(self.selectionChanged)
 		self.parent["config"].onSelectionChanged.remove(self.selectionChanged)
 
 	def selectionChanged(self):
-- 
2.1.4

